ide/coqide/wg_MessageView.ml

```diff
@@ -38,23 +38,35 @@ class type message_view =
     method push : Ideutils.logger
       (** same as [add], but with an explicit level instead of [Notice] *)
 
+    (** Callback for the Ltac debugger *)
+    method debug_prompt : Pp.t -> unit
+
     method has_selection : bool
     method get_selected_text : string
   end
 
+let forward_send_db_cmd = ref ((fun x -> failwith "forward_send_db_cmd")
+    : string -> unit)
+
+(* The buffer can contain prompt or feedback messages *)
+type message_entry_kind =
+  | Fb of Feedback.level * Pp.t
+  | Prompt of Pp.t
+
 let message_view () : message_view =
   let buffer = GSourceView3.source_buffer
     ~highlight_matching_brackets:true
     ~tag_table:Tags.Message.table
     ?style_scheme:(style_manager#style_scheme source_style#get) ()
   in
   let mark = buffer#create_mark ~left_gravity:false buffer#start_iter in
+  let _ = buffer#create_mark ~name:"end_of_output" buffer#end_iter in
   let box = GPack.vbox () in
   let scroll = GBin.scrolled_window
     ~vpolicy:`AUTOMATIC ~hpolicy:`AUTOMATIC ~packing:(box#pack ~expand:true) () in
   let view = GSourceView3.source_view
     ~source_buffer:buffer ~packing:scroll#add
-    ~editable:false ~cursor_visible:false ~wrap_mode:`WORD ()
+    ~editable:false ~cursor_visible:false ~wrap_mode:`CHAR ()
```

Comment:
I definitely don't want this change, it's very annoying for readability to get the output cut at random places.

Reply:
WORD gives unusual behavior for terminal input, but I will change it as you suggest.  The current inputs are very short, so users are unlikely to run into it.  Here are the before and after when I type "Y":

![image](https://user-images.githubusercontent.com/1253341/116006514-7333a380-a5c0-11eb-8e4c-cf908a45d56b.png)

![image](https://user-images.githubusercontent.com/1253341/116006522-7f1f6580-a5c0-11eb-8ae8-2f6173640fc6.png)

I expect things sent as Pp.t would end up word-wrapped with either setting, but some messages are just strings, e.g. `Print Grammar constr`.

Reply:
My main argument is that your PR shouldn't affect people not using the debugger. Since the debugger widget is likely to be reimplemented in a near future, there is no point in trying to make it better right now while negatively affecting the basic behaviour of the message window.